migration-guides/1.x-2.0.adoc

```diff
@@ -0,0 +1,205 @@
+https://github.com/jqassistant-archive[]:: Projects and plugins around jQAssistant that are no longer actively maintained
+
+== Migration Steps
+
+=== Upgrade to Java 11 or newer
+
+Java 8 is no longer support, so for executing jQAssistant at least Java 11 is required now.
+
+
+=== Java 17 Setup
+
+If Java 16 or newer is used then additional Java options must be provided to the JVM:
+
+.Maven
+----
+// Windows
+set MAVEN_OPTS=--add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED
+
+// Linux
+export MAVEN_OPTS=--add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED
+----
+
+An alternative option is to place these settings into a file `.mvn/jvm.config` in the root Maven module:
+
+[source]
+..mvn/jvm.config
+----
+--add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED
+----
+
+.Command Line Utility (CLI)
+----
+// Windows
+set JQASSISTANT_OPTS=--add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED
+
+// Linux
+export JQASSISTANT_OPTS=--add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED
+----
+
+=== Migrate jQAssistant Configuration
+
+All configuration options have been __moved__ to the mechanism based on `.jqassistant.yml` file, environment variables and system properties.
+
+The Maven plugin now supports only a very small set of configuration options in the pom.xml, mainly for locating configuration (.jqassistant.yml) files. The same holds for the command line utility.
+
+Previously used configuration options should be migrated to the according options in a `.jqassistant.yml` file located in the root directory of the project to be analyzed. The set of supported options is documented in the https://jqassistant.github.io/jqassistant/doc/2.0.0/manual/#core:Configuration[manual].
+
+NOTE: Existing options can easily be mapped to the new options by their name. The only notable exception is `jqassistant.storeLifeCycle=REACTOR|MODULE` of the Maven plugin which has been renamed and converted to `jqassistant.maven.reuse-store=true|false`.
+
+=== Declare Plugins In Configuration File
+
+The set of plugins provided with the core distributions has been reduced to Java, JUnit, JSON, XML and Maven 3.
+
+Other plugins have been moved to the new https://github.com/jqassistant-plugin/[jQAssistant Plugin organization] on GitHub and must be declared explicitly in the configuration:
+
+[source,yaml]
+.jqassistant.yml
+----
+jqassistant:
+  plugins:
+    - group-id: org.jqassistant.plugin
+      artifact-id: jqassistant-spring-plugin
+      version: 2.0.0
+    - group-id: org.jqassistant.plugin
+      artifact-id: jqassistant-plantuml-report-plugin
+      version: 2.0.1
+----
+
+The following plugins have been removed from the core distributions and must now be declared explicitly:
+
+[options=header]
```

Comment:
Can you put all these plugins in the YAML, so that it is easier to copy? The plugin name and url of the repo may be put there as a comment